sidecar/supervisor/appsupervisor.go

```diff
@@ -21,52 +21,73 @@ import (
 	"strings"
 	"syscall"
 
+	"sync"
+	"time"
+
 	log "github.com/Sirupsen/logrus"
+	"github.com/amalgam8/amalgam8/sidecar/config"
 	"github.com/amalgam8/amalgam8/sidecar/register"
 )
 
 // AppSupervisor TODO
 type AppSupervisor struct {
-	agent *register.RegistrationAgent
-	cmds  []*exec.Cmd
+	agent   *register.RegistrationAgent
+	App     *exec.Cmd
+	Helpers []*exec.Cmd
 }
 
-// AddProcess TODO
-func (a *AppSupervisor) AddProcess(cmdArgs, env []string) {
-	cmd := exec.Command(cmdArgs[0], cmdArgs[1:]...)
-	if a.cmds == nil {
-		a.cmds = make([]*exec.Cmd, 0)
+// NewAppSupervisor TODO
+func NewAppSupervisor(conf *config.Config) *AppSupervisor {
+	a := AppSupervisor{
+		Helpers: []*exec.Cmd{},
 	}
 
-	cmd.Stdin = os.Stdin
-	cmd.Stdout = os.Stdout
-	cmd.Stderr = os.Stderr
+	for _, cmd := range conf.Commands {
+		osCmd := exec.Command(cmd.Cmd[0], cmd.Cmd[1:]...)
+
+		osCmd.Stdin = os.Stdin
+		osCmd.Stdout = os.Stdout
+		osCmd.Stderr = os.Stderr
+
+		cmdEnv := os.Environ()
+		if cmd.Env != nil {
+			cmdEnv = append(cmdEnv, cmd.Env...)
```

Comment:
It does. The idea was that if `cmd.Env` is nil/empty, we leave `osCmd.Env` nil as well, and save some strings processing, etc when starting the process.